frontend/src/pages/projects/screens/spawner/storage/usePreferredStorageClass.ts

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { AppContext } from '~/app/AppContext';
+import { StorageClassKind } from '~/k8sTypes';
+
+const usePreferredStorageClass = (): StorageClassKind | undefined => {
+  const {
+    dashboardConfig: {
+      spec: { notebookController },
+    },
+    storageClasses,
+  } = React.useContext(AppContext);
+
+  const defaultClusterStorageClasses = storageClasses.filter((storageclass) =>
+    storageclass.metadata.annotations?.['storageclass.kubernetes.io/is-default-class']?.includes(
+      'true',
+    ),
+  );
+
+  const configStorageClassName = notebookController?.storageClassName ?? '';
+
+  if (defaultClusterStorageClasses.length != 0) {
```

Comment:
as of now, this means that is there is a cluster storage class default, that will always be used by returning undefined here. My suggestion is to only let the cluster fill in storageClassName from cluster default if and only if OdhDashBoardConfig notebookController.storageClassName is also undefined.
`  if (defaultClusterStorageClasses.length != 0 && == '' ) {`